auto_changelog/repository.py

```diff
@@ -117,10 +118,12 @@ def _repository_is_empty(self):
         return not bool(self.repository.references)
 
     @staticmethod
-    def _init_commit_tags_index(repo: Repo, tag_pattern: str) -> Dict[Commit, List[TagReference]]:
+    def _init_commit_tags_index(repo: Repo, tag_pattern: str, tag_prefix: str) -> Dict[Commit, List[TagReference]]:
         """ Create reverse index """
         reverse_tag_index = {}
-        for tagref in filter(lambda tagref_: re.search(tag_pattern, tagref_.name), repo.tags):
+        for tagref in filter(
+            lambda tagref_: tagref_.name.startswith(tag_prefix) and re.search(tag_pattern, tagref_.name), repo.tags
```

Comment:
Hello !

Same remark that #55. If we take the default semver regex, with the current version of the code with a prefix `"v"` the tag `"vsomething0.0.0"` is taking into account. Because `re.search` scans through string looking for the first location where the regular expression pattern produces a match. I think this is better to use `re.fullmatch` instead this way we only take tags corresponding to the regex specified.

Reply:
I changed it to the `re.fullmatch`.
Can you please approve/correct my thoughts about how combinations should work?: 

 --tag-pattern | --tag-prefix | result behaviour
--- | --- | ---
 not set | not set | filter semver
 not set | prefix | filter prefix
 pattern | not set | filter pattern
 pattern | prefix | filter prefix and pattern

Reply:
I implemented table above and add more tests. If you have any notes, please let me know, I should be able to work on it tomorrow.

Reply:
Hello !
In my opinion, the combinations should works this way 

| --tag-pattern| --tag-prefix| result behaviour |
| ------------- |:-------------:| -----:|
| not set     | not set | filter semver |
| not set     | prefix | filter prefix and semver |
| pattern    | not set | filter pattern |
| pattern        | prefix  | filter prefix and pattern|

By default the tag pattern is semver and the prefix is "", it seems logic that if we only set a new prefix the default value of default pattern is still valid, so we takes still takes into account semver pattern.

Reply:
This will not work for example when you try to filter the "v" prefix. As v0.1.2 is not valid semver. And I'm cumulating filters there, not concatenating them.

Reply:
I was following to logic implementated at https://www.npmjs.com/package/auto-changelog.

If a prefix is set we check for its presence and them we are checking the rest of the tag to be compliant to the pattern.

For exemple, with the prefix "v", v0.1.2 is valid. We found the prefix which is "v" and then we check if 0.1.2 is a semver version which is ok. 

I implemented this logic in my PR. IMO seems way more intuitive that way.

Reply:
Ok, I will update my code then to follow this.

Reply:
Hello !

Did you have time to change it ?

Thanks a lot

Reply:
Hi, sorry I don't. 
Regarding corona situation we have the emergency state in the company so I spend the majority of my time working and rest of it entertaining kid in the quaranteen.

If nothing will go worst, I should be able to work on it next week.